phpseclib/Net/SSH2.php

```diff
@@ -3693,10 +3694,10 @@ private function get_binary_packet()
             $current = microtime(true);
             $message_number = isset(self::$message_numbers[ord($payload[0])]) ? self::$message_numbers[ord($payload[0])] : 'UNKNOWN (' . ord($payload[0]) . ')';
             $message_number = '<- ' . $message_number .
-                              ' (since last: ' . round($current - $this->last_packet, 4) . ', network: ' . round($elapsed, 4) . 's)';
+                              ' (since last: ' . round($current - $this->last_packet, 4) . ', network: ' . round($packet->read_time, 4) . 's)';
             $this->append_log($message_number, $payload);
-            $this->last_packet = $current;
         }
+        $this->last_packet = microtime(true);
```

Comment:
Use of `microtime(true)` over `$current` in populating `last_packet` discounts any time spent in the logging subroutine.